vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/codec/json/JsonObjectCodec.java

```diff
@@ -106,6 +149,14 @@ protected Object readObjectId(BsonReader reader, DecoderContext ctx) {
 
   @Override
   protected Object readDateTime(BsonReader reader, DecoderContext ctx) {
```

Comment:
This is the format supported by the JSON parser which is part of the BSON library shared by all of the 3.x Mongo Java Drivers -- http://mongodb.github.io/mongo-java-driver/3.0/bson/.  To see its parser in action you do `Document.parse`.

And yes, it is confusing because the formats it supports for the extended JSON constructs are a mixture of "shell" and "strict" mode and they aren't fully documented anywhere.  I figured out the formats above by simply parsing JSON and debugging through the code.  Here is a gist which demonstrates legal and illegal extended date JSON -- https://gist.github.com/sfitts/d8a9c2cb04ca8048470d.

Note that though I've been using Mongo for a few years now I've never needed to go through this exercise because I've never used the extended JSON directly and always just reliedon the fact that the driver will also serialize native Java dates (which I understand are not an option here).